### PR TITLE
added solution for jv-equals-hashcode

### PR DESCRIPTION
**Pull request by Rosinskyi**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -16,6 +18,25 @@ public Rectangle(Integer width, Integer length, String color) {
...
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) o;
+        return Objects.equals(width, rectangle.width)
+                && Objects.equals(length, rectangle.length);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]: what about color?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -16,6 +18,25 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]: equals and hashCode we need to write after getters

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +29,24 @@ public Integer getLength() {
...
+
+        Rectangle rectangle = (Rectangle) o;
+        return Objects.equals(width, rectangle.width)
+                && Objects.equals(length, rectangle.length)
+                && Objects.equals(color, rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(width, length, color);
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: @Rosinskyi let's not use java.util.Objects

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java [hunk 4]
```diff
@@ -27,4 +28,32 @@ public Integer getLength() {
...
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o instanceof Rectangle) {
+            Rectangle rectangle = (Rectangle) o;
+            if (!this.width.equals(rectangle.width)) {
+                return false;
+            }
+            if (!this.length.equals(rectangle.length)) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: @Rosinskyi simplify this if statements

Comment (boroda4436) [reply src/main/java/core/basesyntax/Rectangle.java]: collapse them

Comment (Rosinskyi) [reply src/main/java/core/basesyntax/Rectangle.java]: if i write like `if (!this.width.equals(rectangle.width)) return false;` i'll have checkstyle error 'if' construct must use '{}'s. (40:0) [NeedBraces]

File: src/main/java/core/basesyntax/Rectangle.java [hunk 5]
```diff
@@ -27,4 +28,32 @@ public Integer getLength() {
...
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o instanceof Rectangle) {
+            Rectangle rectangle = (Rectangle) o;
+            if (!this.width.equals(rectangle.width)) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
            return (this.width.equals(rectangle.width) 
            && this.length.equals(rectangle.length)) {
```

**Pull request closed without merging**